pgkit/application/models/replica.py

```diff
@@ -240,7 +240,7 @@ def promote(self):
         self.restart()
 
     def _rename_partial_wal_file(self):
-        wal_location_contents = os.listdir(self.wal_location)
+        wal_location_contents = os.listdir(self.wal_location) if os.path.isdir(self.wal_location) else []
```

Comment:
Is there a case that `self.wal_location` is not a directory?

Reply:
If "use_separate_receivewal_service" is not specified when creating the config, then we will not run setup_wal_receive_service function that will create a directory at wal_location. So if none of our configs have the use_separate_receivewal_service param, then there will be no such directory in our system. 
For example, when testing backup and recovery for v15, and adding the following line to the code:
```
print(f"IS DIR: {os.path.isdir(self.wal_location)}, dir: {self.wal_location}, use_separate_receivewal_service: {self.use_separate_receivewal_service}")
```
I get the following line in the terminal:
```
IS DIR: False, dir: /var/lib/postgresql/wals/15/v15replica, use_separate_receivewal_service: False
```

Reply:
In this case, what is the location of WAL files? Are we not supposed to rename the partial WAL file?
I should investigate this.

Reply:
Instead of checking whether the path is a folder, we can put a condition in the `recovery` method before calling `_rename_partial_wal_file`. The only place that this method should be called is when we are using a separate receivewal service, which you can check with `self.use_separate_receivewal_service`.

Reply:
Condition was placed before calling the `_rename_partial_wal_file` function.